ecs-cli/modules/utils/compose/ecs_params_reader_test.go

```diff
@@ -404,3 +405,257 @@ func TestConvertToECSNetworkConfiguration_NoNetworkConfig(t *testing.T) {
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	_, err = ReadECSParams(ecsParamsFileName)
+
+	assert.Error(t, err, "Expected error reading ecs-params: healthcheck test and command can not both be specified")
+}
+
+func TestReadECSParams_WithHealthCheck_IntFieldsBlank(t *testing.T) {
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    mysql:
+      healthcheck:
+        test: ["CMD", "curl", "-f", "http://localhost"]`
+
+	mysqlExpectedHealthCheck := ecs.HealthCheck{
+		Command:     aws.StringSlice([]string{"CMD", "curl", "-f", "http://localhost"}),
+		Interval:    nil,
+		Timeout:     nil,
+		Retries:     nil,
+		StartPeriod: nil,
+	}
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+
+	if assert.NoError(t, err) {
+		taskDef := ecsParams.TaskDefinition
+
+		containerDefs := taskDef.ContainerDefinitions
+		assert.Equal(t, 1, len(containerDefs), "Expected 1 containers")
+
+		mysql := containerDefs["mysql"]
+
+		verifyHealthCheck(t, mysqlExpectedHealthCheck, ecs.HealthCheck(*mysql.HealthCheck))
+	}
+}
+
+func TestReadECSParams_WithHealthCheck_TestFieldBlank(t *testing.T) {
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    mysql:
+      healthcheck:
+        start_period: 10`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+
+	if assert.NoError(t, err) {
+		taskDef := ecsParams.TaskDefinition
+
+		containerDefs := taskDef.ContainerDefinitions
+		assert.Equal(t, 1, len(containerDefs), "Expected 1 containers")
+
+		mysql := containerDefs["mysql"]
+
+		assert.Len(t, mysql.HealthCheck.Command, 0, "Expected health check command to be empty")
+		assert.Nil(t, mysql.HealthCheck.Interval, "Expected interval to be nil")
+		assert.Nil(t, mysql.HealthCheck.Timeout, "Expected timeout to be nil")
+		assert.Nil(t, mysql.HealthCheck.Retries, "Expected retries to be nil")
+		assert.Equal(t, aws.Int64(10), mysql.HealthCheck.StartPeriod)
```

Comment:
why not set up the expected values as an ecs.HealthCheck and reuse verifyHealthCheck() here (like in TestReadECSParams_WithHealthCheck_IntFieldsBlank) ? (also sorry for missing this on my first pass)